### PR TITLE
Merge feature/pit branch to main

### PR DESCRIPTION
**Pull request by manasvinibs**
### Description
Merge PIT changes to main

### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
 - [x] New functionality has javadoc added.
 - [ ] New functionality has a user manual doc added.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/sql/common/setting/Settings.java [hunk 1]
```diff
@@ -23,6 +23,7 @@ public enum Key {
     SQL_SLOWLOG("plugins.sql.slowlog"),
     SQL_CURSOR_KEEP_ALIVE("plugins.sql.cursor.keep_alive"),
     SQL_DELETE_ENABLED("plugins.sql.delete.enabled"),
+    SQL_PAGINATION_API_SEARCH_AFTER("plugins.sql.pagination.api"),
```
Comment (ykmr1224) [inline common/src/main/java/org/opensearch/sql/common/setting/Settings.java:26]: Do we need documentation for this setting?

File: integ-test/src/test/java/org/opensearch/sql/legacy/JoinIT.java [hunk 2]
```diff
@@ -288,6 +288,8 @@ public void hintMultiSearchCanRunFewTimesNL() throws IOException {
     Assert.assertThat(hits.length(), equalTo(42));
   }
 
+  // TODO: Fix joinWithGeoIntersectNL test when SQL_PAGINATION_API_SEARCH_AFTER is true
```
Comment (ykmr1224) [inline integ-test/src/test/java/org/opensearch/sql/legacy/JoinIT.java:291]: Can we leave this TODO item for now?

Comment (vamsi-amazon) [reply integ-test/src/test/java/org/opensearch/sql/legacy/JoinIT.java:291]: Any reason for this? can we capture it in an issue if not possible now?

**Comment by vamsi-amazon** [comment]
LGTM..address the documentation comment and I will approve.

File: common/src/main/java/org/opensearch/sql/common/setting/Settings.java (hunk 1 above)
Comment (manasvinibs) [reply common/src/main/java/org/opensearch/sql/common/setting/Settings.java:26]: Added documentation.

File: integ-test/src/test/java/org/opensearch/sql/legacy/JoinIT.java (hunk 2 above)
Comment (manasvinibs) [reply integ-test/src/test/java/org/opensearch/sql/legacy/JoinIT.java:291]: I have created this issue https://github.com/opensearch-project/sql/issues/2938 to fix this as follow-up.

**Review by vamsi-amazon — APPROVED** [review]

**Review by ykmr1224 — APPROVED** [review]

**Pull request merged**